src/Api/Extensions/AddMagicLinkLimiterBootstrap.cs

```diff
@@ -0,0 +1,86 @@
+using System.Text;
+using System.Text.Json;
+using System.Threading.RateLimiting;
+using Microsoft.Extensions.Caching.Distributed;
+using Passwordless.Service.MagicLinks;
+using Passwordless.Service.Models;
+
+namespace Passwordless.Api.Extensions;
+
+public static class AddMagicLinkLimiterBootstrap
+{
+    public const string MagicLinkRateLimiterPolicyName = "MagicLinkLimiter";
+
+    public static IServiceCollection AddMagicLinks(this IServiceCollection serviceCollection) =>
+        serviceCollection
+            .AddTransient<MagicLinkService>();
+
+    public static IServiceCollection AddMagicLinksLimiter(this IServiceCollection serviceCollection) =>
+        serviceCollection.AddRateLimiter(limiters =>
+            limiters.AddPolicy(MagicLinkRateLimiterPolicyName, context =>
+                new RateLimitPartition<string>(context.Request.GetTenantName() ?? string.Empty, appId =>
+                    context.RequestServices.GetRequiredService<MagicLinkRateLimiterProvider>().GetRateLimiter(appId))));
+}
+
+public class ApplicationRiskMiddleware(RequestDelegate next)
+{
+    public async Task InvokeAsync(HttpContext context, MagicLinkRateLimiterProvider provider)
+    {
+        var tenantId = context.Request.GetTenantName();
```

Comment:
This will throw an Exception on `/sgdsgdsgsd?bg=324324`

Reply:
Removed